invenio_app_ils/patrons/views.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2019 CERN.
+#
+# invenio-app-ils is free software; you can redistribute it and/or modify it
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""Invenio App ILS Patrons views."""
+
+from flask import Blueprint
+from flask import abort, current_app
+from flask_login import current_user
+from invenio_circulation.search.api import search_by_patron_item_or_document
+from invenio_app_ils.circulation.search import get_loans_aggregated_by_states
+from elasticsearch import VERSION as ES_VERSION
+from invenio_app_ils.permissions import need_permissions
+from webargs import fields
+from webargs.flaskparser import use_args
+lt_es7 = ES_VERSION[0] < 7
+
+
+def get_user_loan_information_blueprint(app):
+    """Add patron views to the blueprint."""
+    blueprint = Blueprint("invenio_app_ils_patrons", __name__,)
+
+    @blueprint.route('/me/loans')
+    @use_args({"document_pid": fields.Str(required=True)})
+    def get_user_information(args):
+        return retrieve_user_information(args["document_pid"])
+    return blueprint
+
+
+def retrieve_user_loans_information(patron_pid, document_pid):
+    """Updates the information about the loans of a concrete document of the currently logged in patron."""
+    active_requested_loan_states = (
+        current_app.config["CIRCULATION_STATES_LOAN_REQUEST"]
+        + current_app.config["CIRCULATION_STATES_LOAN_ACTIVE"]
+    )
+    past_loan_states = (
+        current_app.config["CIRCULATION_STATES_LOAN_COMPLETED"]
+    )
+
+    user_information = {
+        "user_has_request": False,
+        "user_has_loan": False,
+        "user_last_loan": None,
+    }
+
+    loans_search = get_loans_aggregated_by_states(document_pid, active_requested_loan_states+past_loan_states, patron_pid)
+    # No need for the loan hits
+    loans_search = loans_search[:0]
```

Comment:
code duplication with `document_circulation.py` does it make sense to extract this into some helper function ? ping @ntarocco

Reply:
not sure it is exactly duplicated, he is re-using the function `get_loans_aggregated_by_states`, it is just that in the `$ref` we need just the count, here we need the details...